### ISSUE TITLE
symmetric group algebra dft gives ZeroDivisionError when p|n!

### ISSUE THREAD

**Issue by jacksonwalters** [sagemath/sage#37751]
### Steps To Reproduce

`SymmetricGroupAlgebra(GF(3),3).dft()`

### Expected Behavior

When $p|n$, this should return an $n! \times n!$ matrix with values in `GF(p)`. 

This is a change of basis matrix from the usual basis of $S_n$ to a direct sum of blocks, which are two-sided ideals $F_p[S_n]e_i$, where the $e_i$ are central primitive orthogonal idempotents. This is also known as the Pierce decomposition.

This should generalize the case p does not divide n, where we obtain a sum of endomorphism algebras $\text{End}(S^\lambda)$ for Specht modules $S^\lambda$ via Maschke's theorem. $F_p$ is not algebraically closed, but I believe the group algeabra still decomposes as a product of matrix rings over division rings over $F_p$, but by Wedderburn's little theorem this must be a field, which would be $F_q$ where $q=p^r$.

### Actual Behavior

`ZeroDivisionError`

This seems to arise from trying to reduce denominators the semi-normal basis. This is covered in _Murphy, Idempotents of the Symmetric Group and Nakayama's Conjecture, '83_.

### Additional Information

https://github.com/sagemath/sage/pull/37748

### Environment

```markdown
- **OS**: Mac OS 14.4.1 Sonoma, M1, 2020
- **Sage Version**: 10.1, from binary
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by jacksonwalters** [comment]
https://github.com/sagemath/sage/pull/37757

**Comment by jacksonwalters** [comment]
https://github.com/sagemath/sage/pull/37757

**Comment by S17A05** [comment]
Fixed in 10.4beta4 via #37757.
